### ISSUE TITLE
sp-timestamp pulls in wasm-timer which requires std

### ISSUE THREAD

**Issue by Recmo** [paritytech/substrate#5607]
The `sp-timestamp` package uses `wasm-timer`:
```toml
[dependencies]
sp-api = { version = "2.0.0-alpha.5", default-features = false, path = "../api" }
sp-std = { version = "2.0.0-alpha.5", default-features = false, path = "../std" }
sp-runtime = { version = "2.0.0-alpha.5", default-features = false, path = "../runtime" }
codec = { package = "parity-scale-codec", version = "1.3.0", default-features = false, features = ["derive"] }
sp-inherents = { version = "2.0.0-alpha.5", default-features = false, path = "../inherents" }
impl-trait-for-tuples = "0.1.3"
wasm-timer = "0.2"
```
Source: <https://github.com/paritytech/substrate/blob/master/primitives/timestamp/Cargo.toml>

The `wasm-timer` crate pulls in `futures`and other crates with default features (the `impl-trait-for-tuples` crate is a proc-macro crate and works if you use cargo nightly `-Zfeatures=all`):
```toml
[dependencies]
futures = "0.3.1"
parking_lot = "0.9"
pin-utils = "0.1.0-alpha.4"
```
Source: <https://github.com/tomaka/wasm-timer/blob/master/Cargo.toml>

This includes `futures` with default features, which includes `futures/std`.

In addition, this globally sets the `futures` and `futures/std` features, which makes many substrate crates include the optional `futures` crate with `std` enabled.

Basically `wasm-time` does not support no-std build and `sp-timestamp` can not unconditionally depend on it.

**Comment by Recmo** [comment]
`sp-timestamp` only ever uses `wasm-timer` in functions which are gated by the `std` flag, so the following looks like a fix:

```diff
diff --git a/primitives/timestamp/Cargo.toml b/primitives/timestamp/Cargo.toml
index cd31c1db2..3c44db586 100644
--- a/primitives/timestamp/Cargo.toml
+++ b/primitives/timestamp/Cargo.toml
@@ -15,7 +15,7 @@ sp-runtime = { version = "2.0.0-alpha.5", default-features = false, path = "../r
 codec = { package = "parity-scale-codec", version = "1.2.0", default-features = false, features = ["derive"] }
 sp-inherents = { version = "2.0.0-alpha.5", default-features = false, path = "../inherents" }
 impl-trait-for-tuples = "0.1.3"
-wasm-timer = "0.2"
+wasm-timer = { version = "0.2", optional = true }
 
 [features]
 default = [ "std" ]
@@ -25,4 +25,5 @@ std = [
        "sp-runtime/std",
        "codec/std",
        "sp-inherents/std",
+       "wasm-timer",
 ]
```
